src/pyxplor/plot_binary.py

```diff
@@ -47,13 +56,149 @@ def plot_binary(
         
-
     Examples
     --------
     binary = ["binary_var1", "binary_var2"]
-    plot_binary(df, binary, "h", 30, 10, "count", 10, 10, "sample.png")
+    fig, ax = plot_binary(df, binary, "count", 1, 1, "h", (10, 10), True)
     """
-    
-    pass # Implementation to be added
\ No newline at end of file
+
+    # Check input_df is a dataframe
+    if not isinstance(input_df, pd.DataFrame):
+        raise ValueError("input_df must be a pandas DataFrame.")
+
+    # Check if list_of_variables are all in the dataframe
+    missing_variables = [var for var in list_of_variables if var not in input_df.columns]
+    if missing_variables:
+        raise ValueError(f"The following variables are not present in the DataFrame: {', '.join(missing_variables)}.")
+
+    # Check if list_of variables are all binary
+    for var in list_of_variables:
+        unique_values = input_df[var].unique()
+        if len(unique_values) != 2 or set(unique_values) - {0, 1}:
+            raise ValueError(f"Variable '{var}' is not binary.")
+
+    # Check plot_kind is either 'count' or 'pie'
+    valid_plot_kinds = {'count', 'pie'}
+    if plot_kind not in valid_plot_kinds:
+        raise ValueError("Invalid value for 'plot_kind'. It should be either 'count' or 'pie'.")
+
+    # Check if label_offset is a number
+    if not (isinstance(label_offset, (int, float))):
+        raise ValueError("label_offset must be a number (integers or floats).")
+
+    # Check if label_fontsize is a number
+    if not (isinstance(label_fontsize, (int, float))):
+        raise ValueError("label_fontsize must be a number (integers or floats).")
+
+    # Check if plot_orientation is either 'h' or 'v'
+    valid_plot_orientations = {'h', 'v'}
+    if plot_orientation not in valid_plot_orientations:
+        raise ValueError("Invalid value for 'plot_orientation'. It should be either 'h' or 'v'.")   
+
+    # Check figsize is a tuple of 2 numbers
+    if not (isinstance(figsize, tuple) and len(figsize) == 2 and
+            isinstance(figsize[0], (int, float)) and isinstance(figsize[1], (int, float))):
+        raise ValueError("figsize must be a tuple of exactly two numbers (integers or floats).")
+            
+    # Check whether output is a boolean value
+    if not isinstance(output, bool):
+        raise ValueError("Output must be a boolean value.")
+
+    # Check whether super_title is a string
+    if not isinstance(super_title, str):
+        raise ValueError("super_title must be a string.")
+
+    # Check whether super_title_font is a number
+    if not isinstance(super_title_font, (int, float)):
+        raise ValueError("super_title_font must be a number (integer or float).")
+
+            
+    # Calculate dimension of subplot
+    total_plots = len(list_of_variables)
+    rows = math.ceil(math.sqrt(total_plots))
+    cols = math.ceil(total_plots / rows)
+
+    # Instantiate fig and ax
+    fig, ax = plt.subplots(rows, cols, figsize=figsize)
+    ax = ax.flatten()
```

Comment:
This line throws an error if ax only has one element. I added the following to mitigate this!

```suggestion
    if len(list_of_variables) > 1:
        ax = ax.flatten()
```